logger/logger_config.go

```diff
@@ -37,21 +39,48 @@ const (
 	TraceLevel Level = "trace"
 )
 
-type Config struct {
-	Level Level
+func (l Level) String() string {
+	return string(l)
 }
 
-func NewConfig() Config {
-	return Config{
-		Level: InfoLevel,
+func (l *Level) UnmarshalText(b []byte) error {
```

Comment:
Are these (`UnmarshalText` / `MarshalText`) necessary to implement JSON serialization/deserialization?

Reply:
Those are not required for JSON serialization. I've put them to support YAML serialization and other serializers which don't realze `logger.Level` is in fact a `string` but that doesn't work at least for the YAML serializer I've used. So removed those functions at: https://github.com/hazelcast/hazelcast-go-client/pull/585/commits/cf7f17de479a76cb0795726025e896681cb0b265